File: Português/meu aprendisado/linguagem c/exercises/modulo1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float get_pos_num();
int cash(float money);

int main(void)
{
    float money = get_pos_num();
    printf("%i\n", cash(money));
}

float get_pos_num(void)
{
    float n;
    do
    {
        n = get_float("Troco: ");
    }
    while (n < 0.0);
    //n = round(n * 100);
    return n;
}

int cash(float money)
{
    int count = 0;
    while (0.0 < money)
    {
        if (0.25 <= money)
        {
            count++;
            money -= 0.25;
        }
        else if (0.10 <= money)
        {
            count++;
            money -= 0.10;
        }
        else if (0.5 <= money)
        {
            count++;
            money -= 0.5;
        }
        else
        {
            count++;
            money -= 0.1;
        }
    }
    return count;
}

File: Português/meu aprendisado/cs50/exercises/modulo6/cash/cash.py
from cs50 import get_float
from math import floor

def main():
    money = get_pos_float("Change owed: ")
    print(f"{cash(money)}")


def get_pos_float(text:str) -> float:
    height = get_float(text)
    while 0.00 >= height or height > 100.00:
        height = get_float(text)
    return floor(height * 100)


def cash(money:float) -> int:
    count = 0
    while money > 0:
        if money >= 25:
            money -= 25

        elif money >= 10:
            money -= 10

        elif money >= 5:
            money -= 5

        elif money >= 1:
            money -= 1
        count += 1
    return count

main()
